src/main/java/com/blackducksoftware/integration/hub/alert/web/actions/ConfigActions.java

```diff
@@ -63,26 +64,68 @@ public boolean doesConfigExist(final Long id) {
             if (foundEntity != null) {
                 final R restModel = objectTransformer.databaseEntityToConfigRestModel(foundEntity, configRestModelClass);
                 if (restModel != null) {
-                    return Arrays.asList(restModel);
+                    final R maskedRestModel = maskRestModel(restModel);
+                    return Arrays.asList(maskedRestModel);
                 }
             }
             return Collections.emptyList();
         }
         final List<D> databaseEntities = repository.findAll();
         final List<R> restModels = objectTransformer.databaseEntitiesToConfigRestModels(databaseEntities, configRestModelClass);
-        return restModels;
+        return maskRestModels(restModels);
+    }
+
+    public R maskRestModel(final R restModel) {
```

Comment:
This gets overridden if needed by the actions to hide fields. We could make this abstract and force all actions to implement, ideas?

Reply:
I think making it abstract works, but we should also consider how we are using inheritance. We have been abstracting a lot of functionality that is used in multiple places, but we should really decide whether the functionality makes sense as part of a class hierarchy, or as a separate component. Just food for thought.